modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt

```diff
@@ -75,35 +81,89 @@ class GolandRunConfigurationExtension : GoRunConfigurationExtension() {
     ) {
         val service = configuration.getProject().service<MirrordProjectService>()
 
-        if (commandLineType == GoRunningState.CommandLineType.RUN &&
-            service.enabled &&
-            SystemInfo.isMac &&
-            state.isDebug
-        ) {
-            val delvePath = getCustomDelvePath()
-            // convert the delve file to an executable
-            val delveExecutable = Paths.get(delvePath).toFile()
-            if (delveExecutable.exists()) {
-                if (!delveExecutable.canExecute()) {
-                    delveExecutable.setExecutable(true)
-                }
-                // parameters returns a copy, and we can't modify it so need to reset it.
-                val patchedParameters = executor.parameters
-                for (i in 0 until patchedParameters.size) {
-                    // no way to reset the whole commandline, so we just remove each entry.
-                    executor.withoutParameter(0)
-                    val value = patchedParameters[i]
-                    if (value.toPresentableString().endsWith("/dlv", true)) {
-                        patchedParameters[i] = PathParameter(delveExecutable.toString())
-                    }
-                }
-                executor.withParameters(*patchedParameters.toTypedArray())
+        if (commandLineType != GoRunningState.CommandLineType.RUN || !service.enabled || !SystemInfo.isMac || !state.isDebug) {
+            super.patchExecutor(configuration, runnerSettings, executor, runnerId, state, commandLineType)
+            return
+        }
+
+        val ourDelvePath = getCustomDelvePath()
+        val delveExecutable = Paths.get(ourDelvePath).toFile()
+        if (!delveExecutable.exists()) {
+            MirrordLogger.logger.error("Failed to find delve bundled with mirrord plugin ($ourDelvePath)")
+            super.patchExecutor(configuration, runnerSettings, executor, runnerId, state, commandLineType)
+            return
+        }
+
+        if (!delveExecutable.canExecute()) {
+            delveExecutable.setExecutable(true)
+        }
+
+        val ourDelveVersion = try {
+            getDelveVersion(ourDelvePath)
+        } catch (e: Throwable) {
+            MirrordLogger.logger.error("Failed to get version delve bundled with mirrord plugin ($ourDelvePath)", e)
```

Comment:
Is there a possobility to enter this catch from a "correct" flow? like should there be any reason for the version to fail on our shipped dlv other then somthing unexpected (unsupported system ect...)

Reply:
Maybe some weird remote execution contexts, but I'm not sure. In most scenarios, this failure would mean that our build logic is buggy. Do you think an exception would be better?

Reply:
Fixed